File: backend/src/infrastructure/repository/articles.repository.impl.ts
import { ArticleDatasource } from '../../domain/datasources/article.datasource';
import { ArticleEntity } from '../../domain/entities/article.entity';
import { ArticleRepository } from '../../domain/repository/article.repository';

export class ArticlesRepositoryImpl implements ArticleRepository {

    constructor(
        private readonly articleDatasource: ArticleDatasource
    ) {}

    async saveArticles(articles: ArticleEntity[]): Promise<void> {
        this.articleDatasource.saveArticles(articles);
    }

    async getArticles(): Promise<ArticleEntity[]> {
        return this.articleDatasource.getArticles();
    }

}

File: frontend/lib/infrastructure/repositories/articles_repository_impl.dart
import 'package:frontend/domain/datasources/articles_datasource.dart';
import 'package:frontend/domain/entities/article.dart';
import 'package:frontend/domain/repositories/articles_repository.dart';


class ArticlesRepositoryImpl extends ArticlesRepository {

  final ArticlesDatasource articlesDatasource;

  ArticlesRepositoryImpl(this.articlesDatasource);

  @override
  Future<List<Article>> getArticles() {
    return articlesDatasource.getArticles();
  }

}